### ISSUE TITLE
haxe.Template conditions

### ISSUE THREAD

**Issue by filt3rek** [HaxeFoundation/haxe#8560]
Hej all ,

I use haxe.Template for years now, and never really knew how to write conditions because sometimes It complains that it miss parenthisis, sometimes not, sometimes it complains for a space between a test, sometimes no...
You can see it here [https://try.haxe.org/#011d7](https://try.haxe.org/#011d7) : 
```haxe
class Test {
    static function main() {
        var ctx	= {
            name : "Foo",
            isMale : false,
            len : 5
        }
        var s1	= '::if name=="Foo"::Ok::else::No::end::';			// fails
        var s2	= '::if (name=="Foo")::Ok::else::No::end::';			// Ok
        var s3	= '::if(name == "Foo")::Ok::else::No::end::';			// fails
        var s4	= '::if ((name=="Foo")&&(isMale))::Ok::else::No::end::';	// Ok
        var s5	= '::if ((name=="Foo") && (isMale))::Ok::else::No::end::';	// fails

        var s6	= '::if len>4::Ok::else::No::end::';	// fails
        var s7	= '::if (len>4)::Ok::else::No::end::';	// Ok
        var s8	= '::if(len>4)::Ok::else::No::end::';	// fails
      
        var t = new haxe.Template( s8 );
        	trace( t.execute( ctx ) );
    }
}
```

Thanks for reading,
Regards,
Michal

**Comment by RealyUniqueName** [comment]
Spaces are fixed now.
But "complex" expressions like `len > 4` should always be enclosed in parentheses.

**Comment by filt3rek** [comment]
Ths Aleksandr, 
Since when "spaces" are fixed please ? 3.4.4 is not

**Comment by RealyUniqueName** [comment]
Since one hour ago :)
I doubt there will be any patch releases for Haxe 3.4.* branch. 
But you can just copy the fixed version of `haxe.Template` to your project: https://github.com/HaxeFoundation/haxe/blob/dbc9e9432736747293d0c6648bdb7759e9d0453c/std/haxe/Template.hx
It should be compatible with Haxe 3.4
